src/core/server/http/router/request.ts

```diff
@@ -184,6 +190,8 @@ export class KibanaRequest<
 
     const options = ({
       authRequired: request.route.settings.auth !== false,
+      // some places in LP call KibanaRequest.from(request) manually. remove fallback to true before v8
+      xsrfRequired: (request.app as KibanaRouteState)?.xsrfRequired || true,
```

Comment:
should this be `(request.app as KibanaRouteState)?.xsrfRequired ?? true` ATM I think this will always resolve to `true` ?

Reply:
right, need to fix. I'm spoiled by TS type checks